types/node/fs.d.ts

```diff
@@ -684,6 +684,7 @@ declare module "fs" {
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
      */
     function rmdir(path: PathLike, callback: (err: NodeJS.ErrnoException | null) => void): void;
+    function rmdir(path: PathLike, options: { emfileWait?: number; maxBusyTries?: number; recursive?: boolean; }, callback: (err: NodeJS.ErrnoException | null) => void): void;
```

Comment:
How about breaking out the options into an interface, just like was done with MakeDirectoryOptions for mkdir?

(I just submitted a duplicate of the PR by mistake: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39118. I added an interface called RemoveDirectoryOptions.)

Reply:
To be consistent with the other declarations, we should perhaps add a separate jsdoc comment for each overload?

Reply:
> How about breaking out the options into an interface?

Agree. This was also suggested in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39105#issuecomment-541788723

Reply:
I extracted an interface and put it before the first `rmdir` definition. Is it alright the way I did it?

Reply:
> To be consistent with the other declarations, we should perhaps add a separate jsdoc comment for each overload?

How should I go about that? Should I copy the annotation from above and adjust it?

Reply:
At the moment the options argument is missing a jsdoc comment.

fs.mkdir recently received a similar `recursive` flag. That code looks like this:
```
/**
 * Asynchronous mkdir(2) - create a directory.
 * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
 * @param options Either the file mode, or an object optionally specifying the file mode and whether parent folders
 * should be created. If a string is passed, it is parsed as an octal integer. If not specified, defaults to `0o777`.
 */
function mkdir(path: PathLike, options: number | string | MakeDirectoryOptions | undefined | null, callback: (err: NodeJS.ErrnoException | null) => void): void;

/**
 * Asynchronous mkdir(2) - create a directory with a mode of `0o777`.
 * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
 */
function mkdir(path: PathLike, callback: (err: NodeJS.ErrnoException | null) => void): void;
```

That is, one declaration with the options argument and one without. I was thinking it'd nice to make this declaration consistent with fs.mkdir.